### PR TITLE
Provide detailed errors

### PR DESCRIPTION
**Pull request by balazssimongen**
Improving error handling and fixing a minor bug related to image upload counting

### CODE REVIEW

File: src/utils/errors/ErrorCodes.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export enum ErrorCodes {
+  CONFIGURER_ERROR,
```
Comment (petergaras) [inline src/utils/errors/ErrorCodes.ts:2]: You could define value for each so you can get rid of that long .toString().toLowerCase() call.
```
export enum ErrorCodes {
  CONFIGURER_ERROR = 'configurer_error',
  ...
}
```

File: src/aggregator/diff-aggregator.spec.ts [hunk 2]
```diff
@@ -765,7 +779,7 @@ describe('DiffAggregator', () => {
           });
         });
 
-        it('should throw error', async () => {
+        it('should throw prune error', async () => {
```
Comment (petergaras) [inline src/aggregator/diff-aggregator.spec.ts:782]: should throw configurer error

File: src/salesforce/salesforce-api.ts [hunk 3]
```diff
@@ -11,6 +11,9 @@ import { Image } from '../model';
 import { validateNonNull } from '../utils/validate-non-null.js';
 import { SalesforceAccessTokenResponse } from './model/salesforce-access-token-response.js';
 import { LANGUAGE_MAPPING } from './salesforce-language-mapping.js';
+import { InvalidCredentialsError } from '../adapter/errors/InvalidCredentialsError';
```
Comment (petergaras) [inline src/salesforce/salesforce-api.ts:14]: .js missing

**Comment by petergaras** [comment]
Don't forget to bump version

**Review by tkurucz-genesys — APPROVED** [review]

**Pull request merged**